### PR TITLE
enable Applitools (Eyes) / GitHub integration

### PR DESCRIPTION
**Pull request by sureshc**
Set environment variables that control how the Applitools gem identifies each batch of Eyes tests, to facilitate integration with GitHub so we can enable branch-specific Eyes baselines.

Applitools [requires](https://www.applitools.com/docs/topics/integrations/github-integration.html#Configuration) setting the environment variable `APPLITOOLS_BATCH_ID` to the commit hash of the current build.  It also seems to require that we no longer explicitly set the eyes instance `batch_name` attribute and other attributes like `branch_name`

### CODE REVIEW

**Review by davidsbailey — APPROVED** [review]
Nice progress Suresh! LGTM after patching this onto the test machine and making sure `rake test:eyes_ui` still passes. if it fails, it may be because it is comparing against the staging branch instead of the test branch. so you may need to dig into the docs and find out what the new way of telling (or overriding) applitools which branch it should be comparing against.

**Comment by davidsbailey** [comment]
ohh actually I just realized that in order for eyes tests to pass as described above, you would probably need to do something like this on the test machine (UPDATED):
```
git fetch
git cherry-pick d2f310d1d7842241d4be3fb9350f43db239e3ebc # the commit in this PR
git reset HEAD^ # now you are still on the test branch, with your PR patched in as untracked changes
rake test:eyes_ui
git checkout . # clean up
```
that would make sure that the only thing different between what you're testing and what happened in the eyes test run is what's in this PR. otherwise, there might be some other differences pulled in.

**Review by wjordan — APPROVED** [review]
I don't know enough about this Applitools code to comment on whether this PR will work, but at least nothing concerns me in the diff.

This only runs in our test pipeline so relatively low risk/impact and there's a bit of a chicken/egg situation with testing test-infrastructure code, so I'm fine with you going ahead and merging and/or cherry-picking this diff on the test server however is needed to test this out in our pipeline.

File: dashboard/test/ui/runner.rb [hunk 1]
```diff
@@ -352,17 +352,14 @@ def eyes?
...
-  ENV['BATCH_ID'] = "#{GIT_BRANCH}_#{SecureRandom.uuid}".gsub(/[^\w-]+/, '_')
+  ENV['APPLITOOLS_API_KEY'] = CDO.applitools_eyes_api_key
+  # The Eyes/Applitools gem uniquely identifies a test run by the git commit hash.
+  ENV['APPLITOOLS_BATCH_ID'] = COMMIT_HASH
 end
 
 def applitools_batch_url
   return nil unless eyes?
-  "https://eyes.applitools.com/app/batches/?startInfoBatchId=#{ENV['BATCH_ID']}&hideBatchList=true"
+  "https://eyes.applitools.com/app/batches/?startInfoBatchId=#{ENV['APPLITOOLS_BATCH_ID']}&hideBatchList=true"
```
Comment (islemaster) [inline dashboard/test/ui/runner.rb:366]: Does this batch URL even work anymore?  I think they might have changed their UI and broken these links for us at some point.  Not blocking, but maybe worth looking into.

**Comment by sureshc** [comment]
@davidsbailey I finally got this working!  Can you take another look?

File: dashboard/test/ui/runner.rb [hunk 2]
```diff
@@ -665,6 +654,15 @@ def run_feature(browser, feature, options)
   # BrowserStack was reporting Windows 6.0 and 6.1, causing different baselines
   run_environment['APPLITOOLS_HOST_OS'] = 'Windows 6x' unless browser['mobile']
 
+  # When executing in a CircleCI environment use the hash of the commit that triggered
+  # the CircleCI build.  Our CircleCI build includes a step to merge the staging
+  # branch into the feature branch locally in the CircleCI environment, which means
+  # we can't use the COMMIT_HASH constant, because it's a different commit
+  # than the one at the head of the feature branch in GitHub.  The Applitools / GitHub
+  # integration requires APPLITOOLS_BATCH_ID to be set to the hash of the commit
+  # associated with the Pull Request so that it can lookup information about the source / target branch.
```
Comment (davidsbailey) [inline dashboard/test/ui/runner.rb:688]: great explanation!

**Review by davidsbailey — APPROVED** [review]
Looking forward to seeing this in action!

**Pull request closed without merging**